### ISSUE TITLE
No titleAriaId prop for Panel

### ISSUE THREAD

**Issue by rkalivarapu** [OfficeDev/office-ui-fabric-react#11106]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)

> Note that our screen reader support is limited to Edge + Narrator. Issues that do not reproduce in Edge + Narrator, and aren't caused by obvious invalid aria values, should be filed with the respective screen reading software, not the Fabric repo.

### Describe the issue:

We have titleAriaId property exposed for Modal but not for Panel. Can we have this prop for Panel?
We need this prop as screen readers should read out the dialog title(if we override).

**Comment by JasonGore** [comment]
Are you using `onRenderHeader` prop? It should be setting the same ARIA attributes in order for the header text to be read out.

**Comment by rkalivarapu** [comment]
We are rendering header using onRenderNavigationContent

**Comment by JasonGore** [comment]
`onRenderNavigationContent` callbacks shouldn't be used for rendering headings. `onRenderNavigationContent` doesn't know if header content is provided or which content is the header. You would either have to supply the ARIA properties necessary to define the heading, or use the `headerText` or `onRenderHeader` props to inform `Panel` what the heading content is.

**Comment by dssharma** [comment]
We have been facing similar accessibility issue where screen reader doesn't read out for Panel, when headerText is not given. Hacky way is to have onRenderHeader with <div style="display: none">{Heading to be read out by screen reader}</div>.
We would expect to have some sort of ariaLabel over this component.

**Comment by JasonGore** [comment]
@dssharma In this case I don't think the `aria-label` is sufficient. `Panel` actually uses the ARIA `heading` role to make sure that the header text is communicated as a dialog to the user. Simply using `aria-label` may not dictate the heading when the `Panel` is navigated to, particularly if it's not focusable.

I've just tested and `headerText` is automatically applied to the `heading` role and narrated when the Panel appears:
![image](https://user-images.githubusercontent.com/26070760/69169994-2b951800-0aae-11ea-94a3-98d1e5b3fc0c.png)

In any case, I'm not clear what your use case is. Are you wanting to use `headerText` and have it narrated without displaying the actual text?
